src/sentry/utils/sdk.py

```diff
@@ -124,6 +130,21 @@ def on_full_queue(*args, **kwargs):
     transport._worker.on_full_queue = on_full_queue
 
 
+def traces_sampler(sampling_context):
+    # If there's already a sampled decision, just use that
+    if sampling_context["parent_sampled"] is not None:
+        return sampling_context["parent_sampled"]
+
+    # Resolve the url, and see if we want to set our own sampling
+    if "wsgi_environ" in sampling_context:
+        match = resolve(sampling_context["wsgi_environ"].get("PATH_INFO"))
+        if match and match.url_name in SAMPLED_URL_NAMES:
+            return 1.0
```

Comment:
Chatted with Markus about doing this automatically, and he's not keen on it because he feels like resolving the route is a (relatively) expensive operation, so I stand by my suggestion of only doing it if the sampling decision is no (if it's yes, setting it to 1 is a no-op anyway).

Reply:
So to double check `sampling_context["parent_sampled"]` can be (based on what i'm seeing locally testing this):
- `True`, the transaction decision was already to sample
- `None`, there isn't a sampling decision yet, or the decision was not to sample

So is this current implementation matching your suggestion?

Reply:
The parent decision can also be `False`. But yes, sorry - with the precedence change (to put checking for a parent decision first) the current implementation is correct.